src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/BasePageObject.java

```diff
@@ -43,17 +46,44 @@ public void tapOnElement(WebElement element) {
     jsexec.executeScript("arguments[0].click();", element);
   }
 
-  public void waitMilliseconds(int time, String methodName) {
+  public void waitMilliseconds(int time, String message) {
     try {
       Thread.sleep(time);
     } catch (InterruptedException e) {
-      PageObjectLogging.log(methodName, e.getMessage(), false);
+      PageObjectLogging.log(message, e.getMessage(), false);
+    }
+  }
+
+  /**
+   * First waits for spinner to be visible and then waits for spinner to be hidden
+   */
+  public void waitForLoadingSpinnerToFinishReloadingPage() {
+    try {
+      waitForElementByElement(loadingSpinner);
+      waitForElementPresenceByBy(By.cssSelector(".loading-overlay.hidden"));
+    } catch (TimeoutException e) {
+    }
+  }
+
+  /**
+   * Example: wait 10 sec for element and check each 0.5 for that element (10 / 0.5 = 20 attempts)
+   *
+   * @param element            WebElement
+   * @param timeOutInSec       int
+   * @param checkOutInMilliSec int
+   */
+  public void waitForElementVisibleByElementCustomTimeOut(WebElement element, int timeOutInSec,
+                                                          int checkOutInMilliSec) {
+    WebDriverWait wait = new WebDriverWait(driver, timeOutInSec);
+    driver.manage().timeouts().implicitlyWait(checkOutInMilliSec, TimeUnit.MILLISECONDS);
+    try {
+      wait.until(ExpectedConditions.visibilityOf(element));
+    } finally {
+      restoreDeaultImplicitWait();
     }
   }
 
-  public static void turnOnMercurySkin(WebDriver driver, String wikiURL) {
-    BasePageObject base = new BasePageObject(driver);
-    SpecialMercuryPageObject mercuryPage = base.openSpecialMercury(wikiURL);
-    mercuryPage.clickMercuryButton();
+  public void failTest(boolean fail) {
```

Comment:
Because if i mark test as fail from pageobjectlogging, it still mark tests as success in jenkins. But i want to mark failed functionality and if i can continue executing test and at the end i want to fail test